src/main/java/com/wavefront/datastructures/TraceFromWF.java

```diff
@@ -1,26 +1,79 @@
 package com.wavefront.datastructures;
 
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
+@JsonAutoDetect(fieldVisibility = Visibility.ANY)
 public class TraceFromWF {
   private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
 
-  public String traceId;
-  public long start_ms;
-  public long end_ms;
-  public long total_duration_ms;
-  public List<SpanFromWF> spans;
+  private String traceId;
+  private long start_ms = Long.MAX_VALUE;
+  private long end_ms = Long.MIN_VALUE;
+  private long total_duration_ms;
+  private List<SpanFromWF> spans;
+
+  public List<SpanFromWF> getSpans() {
+    return spans;
+  }
+
+  public void setSpans(List<SpanFromWF> spans) {
+    this.spans = spans;
+    setStartAndEnd();
```

Comment:
I guess the is needed only for non-null(non-empty) case, so could you move it under else statement.